tools/api_proto_breaking_change_detector/detector.py

```diff
@@ -0,0 +1,129 @@
+from abc import ABC, abstractmethod
```

Comment:
Please add a comment that describes what this tool is doing

Reply:
Python style is to put the docstring at the top of the file, before imports. See https://realpython.com/documenting-python-code/#package-and-module-docstrings.